src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -399,6 +354,48 @@ bool ProcessPrimaryQueue ()
 			return true;
 		}
 
+		bool ProcessMarkedPending ()
+		{
+			bool marked = false;
+			foreach (var pending in Annotations.MarkedPending ()) {
+				marked = true;
+
+				// Some pending items might be processed by the time we get to them.
+				if (Annotations.IsProcessed (pending))
+					continue;
+
+				switch (pending) {
+				case TypeDefinition type:
+					MarkType (type, DependencyInfo.AlreadyMarked, type);
+					break;
+				case MethodDefinition method:
+					MarkMethod (method, DependencyInfo.AlreadyMarked, null);
+					// Methods will not actually be processed until we drain the method queue.
```

Comment:
Yup - I actually noticed this because MarkMethod sets the MethodAction to Parse, and that wasn't happening correctly in some cases when using EnqueueMethod.

Reply:
Actually - since this only allows early marking of MethodDefinition, not MethodReference, I don't think it will change the behavior for generics, but it will improve things if we add a MethodReference case.